src/device/pf_fspm.c

```diff
@@ -60,9 +62,94 @@ int pf_fspm_validate_configuration(
       return -1;
    }
 
+   if ((p_cfg->im_0_data.im_supported & im_mask) > 0)
+   {
+      LOG_ERROR(PNET_LOG, "FSPM(%d): The I&M supported setting is wrong. Given: 0x%04x\n", __LINE__, p_cfg->im_0_data.im_supported);
+      return -1;
+   }
+
    return 0;
 }
 
+/**
+ * @internal
+ * Load the I&M settings from nonvolatile memory, if existing.
+ *
+ * @param net              InOut: The p-net stack instance
+ */
+static void pf_fspm_load_im(
+   pnet_t                  *net)
+{
+   pnet_im_nvm_t           file_im;
+   const char              *p_file_directory = NULL;
+
+   (void)pf_cmina_get_file_directory(net, &p_file_directory);
+
+   if (pf_file_load(p_file_directory, PNET_FILENAME_IM, &file_im, sizeof(pnet_im_nvm_t)) == 0)
+   {
+      LOG_DEBUG(PNET_LOG,"FSPM(%d): Did read I&M settings from nvm.\n", __LINE__);
+      memcpy(&net->fspm_cfg.im_1_data, &file_im.im1, sizeof(pnet_im_1_t));
+      memcpy(&net->fspm_cfg.im_2_data, &file_im.im2, sizeof(pnet_im_2_t));
+      memcpy(&net->fspm_cfg.im_3_data, &file_im.im3, sizeof(pnet_im_3_t));
+      memcpy(&net->fspm_cfg.im_4_data, &file_im.im4, sizeof(pnet_im_4_t));
+   }
+   else
+   {
+      LOG_DEBUG(PNET_LOG,"FSPM(%d): Could not yet read I&M settings from nvm. Use values from user configuration.\n", __LINE__);
+   }
+}
+
+/**
+ * @internal
+ * Save the I&M settings to nonvolatile memory, if necessary.
+ *
+ * Compares with the content of already stored settings (in order not to
+ * wear out the flash chip)
+ *
+ * @param net              InOut: The p-net stack instance
+ */
+static void pf_fspm_save_im_if_modified(
+   pnet_t                  *net)
+{
+   pnet_im_nvm_t           output_im;
```

Comment:
```suggestion
   pnet_im_nvm_t           output_im = {0};
```
(ensures that padding bytes are initialised as needed for subsequent memcmp())

Reply:
Done (also initialized current_file_im to be consistent)